src/Magnum/Platform/EmscriptenApplication.cpp

```diff
@@ -562,14 +562,62 @@ EmscriptenApplication::GLConfiguration::GLConfiguration():
 #endif
 
 int EmscriptenApplication::exec() {
-    emscripten_set_main_loop_arg([](void* userData) {
+    redraw();
+    return 0;
+}
+
+void EmscriptenApplication::redraw() {
+    _flags |= Flag::Redraw;
+
+    /* Loop already running, no need to start */
+    if(!!(_flags & Flag::LoopActive)) return;
+
+    // TODO: @mosra There is a race condition potential here,
```

Comment:
@mosra This is up for discussion.

Reply:
What about making `redraw()` really just the flag setter like it was before and move all the rest into a dedicated function that gets explicitly called at the end of every event callback? That way it shouldn't happen that it gets called from inside itself.